python/openassetio/hostAPI/Manager.py

```diff
@@ -1037,57 +1037,83 @@ def createChildContext(self, parentContext):
         @see @ref createContext
         @see @fqref{Context} "Context"
         """
-        return Context(
+        context = Context(
             access=parentContext.access,
             retention=parentContext.retention,
-            locale=parentContext.locale,
-            managerState=self.__impl.createChildState(self.__hostSession,
-                                                      parentContext.managerState)
+            locale=parentContext.locale
         )
 
+        if parentContext.managerState:
+            context.managerState = \
+                self.__impl.createChildState(self.__hostSession, parentContext.managerState)
+
+        return context
+
     @auditApiCall("Session")
-    def freezeContext(self, context):
+    def persistenceTokenForContext(self, context):
         """
-        Freezes the supplied context into a serializable token that
-        can be persisted or distributed between processes to associate
-        subsequent API usage with the supplied context.
+        Returns a serializable token that represents the supplied
+        context's managerState, such that it can be persisted or
+        distributed between processes to associate subsequent API usage
+        with the supplied context.
 
-        The returned token can be passed to @ref thawContext for
-        future API use in another @ref session with the same manager.
+        The returned token can be passed to @ref
+        contextFromPersistenceToken for future API use in another @ref
+        session with the same manager.
 
-        @param context @fqref{Context} "Context" The context to freeze.
+        @param context @fqref{Context} "Context" The context to derive a
+        persistence token for.
 
-        @return `str` The managers frozen state token.
+        @return `str` A persistence token that can be used with @ref
+        contextFromPersistenceToken to create a context associated with
+        the same logical group of actions as the one supplied to this
+        method.
+
+        @warning This only encapsulates the logical identity of the
+        Context, such when restored, any API calls made using the
+        resulting Context will be logically associated with the one
+        supplied here. It does not encode the current access, retention
+        or locale.
 
         @see @ref stable_resolution
         """
-        token = self.__impl.freezeState(context.managerState, self.__hostSession)
+        if not context.managerState :
```

Comment:
[question] I think this makes sense from the perspective of a host that wants to work generically with all managers that may or may not support state tokens, without changing the host code to cover both cases. But I wonder if it is ever valid to use a state token based workflow with managers that don't support it?

Reply:
Capturing offline discussion: we assume that if the (well behaved) manager doesn't provide a `managerState`, then it simply doesn't require one for the workflows that `managerState` (and hence state tokens) is designed to simplify, so its fine for the host to be oblivious.